tests/execution_engine/test_sparkdf_execution_engine.py

```diff
@@ -184,6 +188,157 @@ def test_get_domain_records_with_column_domain(
+                for x in record.tolist()
+            )
+            for record in pd_df.to_records(index=False)
+        ],
+        pd_df.columns.tolist(),
+    )
+    engine = basic_spark_df_execution_engine
+    engine.load_batch_data(batch_id="1234", batch_data=df)
+    data = engine.get_domain_records(
+        domain_kwargs={
+            "column": "a",
+            "row_condition": 'col("a")<2',
+            "condition_parser": "great_expectations__experimental__",
+            "filter_conditions": [
+                RowCondition(
+                    condition="b IS NOT NULL",
+                    condition_type=RowConditionParserType.SPARK_SQL,
+                )
+            ],
+        }
+    )
+
+    expected_column_pd_df = pd_df.iloc[:1]
+    expected_column_df = spark_session.createDataFrame(
+        [
+            tuple(
+                None if isinstance(x, (float, int)) and np.isnan(x) else x
+                for x in record.tolist()
+            )
+            for record in expected_column_pd_df.to_records(index=False)
+        ],
+        expected_column_pd_df.columns.tolist(),
+    )
+
+    assert dataframes_equal(
+        data, expected_column_df
+    ), "Data does not match after getting full access compute domain"
+
+
+def test_get_domain_records_with_different_column_domain_and_multiple_filter_conditions(
+    spark_session, basic_spark_df_execution_engine
+):
+    pd_df = pd.DataFrame(
+        {"a": [1, 2, 3, 4, 5], "b": [2, 3, 4, 5, None], "c": [1, 2, 3, 4, None]}
+    )
+    df = spark_session.createDataFrame(
+        [
+            tuple(
+                None if isinstance(x, (float, int)) and np.isnan(x) else x
+                for x in record.tolist()
+            )
+            for record in pd_df.to_records(index=False)
+        ],
+        pd_df.columns.tolist(),
+    )
+    engine = basic_spark_df_execution_engine
+    engine.load_batch_data(batch_id="1234", batch_data=df)
+    data = engine.get_domain_records(
+        domain_kwargs={
+            "column": "a",
+            "row_condition": 'col("a")<10',
+            "condition_parser": "great_expectations__experimental__",
+            "filter_conditions": [
+                RowCondition(
+                    condition="b IS NOT NULL",
+                    condition_type=RowConditionParserType.SPARK_SQL,
+                ),
+                RowCondition(
+                    condition="NOT isnan(b)",
+                    condition_type=RowConditionParserType.SPARK_SQL,
+                ),
+            ],
+        }
+    )
+
+    expected_column_pd_df = pd_df.iloc[:4]
+    expected_column_df = spark_session.createDataFrame(
+        [
+            tuple(
+                None if isinstance(x, (float, int)) and np.isnan(x) else x
+                for x in record.tolist()
+            )
+            for record in expected_column_pd_df.to_records(index=False)
+        ],
+        expected_column_pd_df.columns.tolist(),
+    )
```

Comment:
Nitpick - seems like this is used a few times. I'm totally cool with verbosity to illustrate how it all works but could this be placed into a fixture or lambda? I think we can leave it as is but just thinking out loud.

Reply:
This is used quite a lot! But a good refactor. Changing now.